frames/src/main/java/com/checkout/frames/component/expirydate/ExpiryDateViewModel.kt

```diff
@@ -27,22 +32,68 @@ internal class ExpiryDateViewModel @Inject constructor(
     private val style: ExpiryDateComponentStyle,
 ) : ViewModel() {
 
-    val componentState = ExpiryDateComponentState(inputComponentStateMapper.map(style.inputStyle))
+    val componentState = provideViewState(style.inputStyle)
     val componentStyle = provideViewStyle(style.inputStyle)
 
+    // Flag to determine that component has been already focused before
+    // Needed to prevent validation on focus switch for initial component state
+    private var wasFocused = false
+
+    fun onFocusChanged(isFocused: Boolean) {
+        if (isFocused) wasFocused = isFocused
+
+        if (!isFocused && wasFocused) {
+            val expiryDate = componentState.expiryDate.value
+            validateExpiryDateOnFocusChanged(expiryDate)
+        }
+    }
+
+    private fun validateExpiryDateOnFocusChanged(expiryDate: String) {
+        if (expiryDate.length >= EXPIRY_DATE_MAXIMUM_LENGTH) {
+            handleValidationResult(cardValidator.validateExpiryDate(expiryDate.take(2), expiryDate.takeLast(2)))
+        } else {
+            // TODO: PIMOB-1401 - Implement error handling for expiry date.
+        }
+    }
+
     fun onExpiryDateInputChange(text: String) {
-        componentState.inputState.inputFieldState.text.value = text
-        handleValidationResult(cardValidator.validateExpiryDate(text, text))
+        val strBuilder = StringBuilder()
+        strBuilder.append(text)
+
+        when (val validateExpiryDateUseCase = ValidateExpiryDateUseCase().execute(strBuilder)) {
```

Comment:
Not sure what stands under a decision to violate DI approach and create the usecase directly.
Could you please help me to figure it out?